### PR TITLE
Bug 3420: Migrate Plugin API module from fx module

### PR DESCRIPTION
**Pull request by ykubota**
This change will migrate `heapstats-plugin-api` module from `heapstats-analyzer` module. Hereby, creating plugin requires only two small modules: `heapstats-core` and `heapstats-plugin-api`.

http://icedtea.classpath.org/bugzilla/show_bug.cgi?id=3420

### CODE REVIEW

**Comment by ykubota** [comment]
* A result of rpmbuild on Fedora 26
```
/home/ykubota/rpmbuild% tree RPMS
RPMS
|-- noarch
|   |-- heapstats-analyzer-2.0.trunk-0.fc26.noarch.rpm
|   `-- heapstats-cli-2.0.trunk-0.fc26.noarch.rpm
`-- x86_64
    |-- heapstats-2.0.trunk-0.fc26.x86_64.rpm
    `-- heapstats-debuginfo-2.0.trunk-0.fc26.x86_64.rpm

2 directories, 4 files
/home/ykubota/rpmbuild% tree /usr/libexec/heapstats/
/usr/libexec/heapstats/
|-- THIRD_PARTY_README
|-- filterDefine.xsd
|-- heapstats-analyzer.jar
|-- heapstats-cli.jar
|-- heapstats.properties
`-- lib
    |-- heapstats-core.jar
    |-- heapstats-jmx-helper.jar
    |-- heapstats-mbean.jar
    |-- heapstats-plugin-api.jar
    `-- jgraphx.jar

1 directory, 10 files
```

File: analyzer/plugin-api/pom.xml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+            <organizationUrl>http://www.icedtea.classpath.org</organizationUrl>
+        </developer>
+    </developers>
+
+    <scm>
+        <url>http://icedtea.classpath.org/hg/heapstats</url>
+        <connection>scm:hg:http://icedtea.classpath.org/hg/heapstats</connection>
+        <developerConnection>scm:hg:ssh://icedtea.classpath.org/hg/heapstats</developerConnection>
+    </scm>
+
```
Comment (YaSuenag) [inline analyzer/plugin-api/pom.xml:60]: I guess this module will be registered to Maven Central.
`pom.xml` in `heapstats-core` has `distributionManagement` tag. Does not need it?

File: analyzer/plugin-api/src/main/java/jp/co/ntt/oss/heapstats/utils/HeapStatsUtils.java [hunk 2]
```diff
@@ -227,19 +226,13 @@ public static void load() throws IOException, HeapStatsConfigException {
...
-        pluginList.add(ThreadRecorderController.class.getPackage().getName());
-        pluginList.add(JVMLiveController.class.getPackage().getName());
-        pluginList.addAll(Arrays.asList(prop.getProperty("plugins", "").split(";")));
-
-        return pluginList.stream()
+        return Arrays.asList(prop.getProperty("plugins", "").split(";"))
+                .stream()
                 .map(s -> s.trim())
                 .filter(s -> s.length() > 0)
                 .distinct()
```
Comment (YaSuenag) [inline analyzer/plugin-api/src/main/java/jp/co/ntt/oss/heapstats/utils/HeapStatsUtils.java]: This statement returns unique `pluginList`. However builtin plugins will be added at `MainWindowController#loadPlugin()`.
`distinct()` should be called at `loadPlugin()`.

File: analyzer/plugin-api/pom.xml (hunk 1 above)
Comment (ykubota) [reply analyzer/plugin-api/pom.xml:60]: Yes, we need it. Thanks for comment.

File: analyzer/plugin-api/src/main/java/jp/co/ntt/oss/heapstats/utils/HeapStatsUtils.java (hunk 2 above)
Comment (ykubota) [reply analyzer/plugin-api/src/main/java/jp/co/ntt/oss/heapstats/utils/HeapStatsUtils.java]: Thanks for comment. I didn't consider the case when a user add built-in plugins by config file, I will fix it.

**Review by YaSuenag — APPROVED** [review]

**Pull request closed without merging**